python/cudf/cudf/_lib/copying.pyx

```diff
@@ -202,17 +218,15 @@ def copy_range(out_col, in_col, int out_begin, int out_end,
     if out_begin > out_end:
         return out_col
 
-    if out_col.null_count == 0 and in_col.has_null_mask:
+    if out_col.null_count == 0 and in_col.mask:
```

Comment:
treating `in_col.mask` as a boolean feels ambiguous, would rather explicitly compare to None:
```suggestion
    if out_col.null_count == 0 and in_col.mask is not None:
```